packages/fxa-settings/src/lib/integrations/integration-factory.ts

```diff
@@ -2,128 +2,293 @@
-      return new WebIntegration();
-    } else if (this.flags.isServiceSync()) {
-      // user is verifying in a different browser.
-      return new SyncBasicIntegration();
-    } else if (this.flags.isServiceOAuth() && isOAuthRelier(this.relier)) {
-      // oauth, user is verifying in a different browser.
-      return new OAuthIntegration(this.relier);
-    }
+  private createPairingAuthorityIntegration(
+    data: ModelDataStore,
+    storageData: ModelDataStore
+  ) {
+    const integration = new PairingAuthorityIntegration(
+      data,
+      storageData,
+      config.oauth
+    );
+    this.initIntegration(integration);
+    return integration;
+  }
+
+  private createPairingSupplicationIntegration(
+    data: ModelDataStore,
+    storageData: ModelDataStore
+  ) {
+    const integration = new PairingSupplicantIntegration(
+      data,
+      storageData,
+      config.oauth
+    );
+    this.initIntegration(integration);
+    this.initClientInfo(integration);
+    return integration;
+  }
+
+  private createOAuthIntegration(
+    data: ModelDataStore,
+    storageData: ModelDataStore
+  ) {
+    // Resolve configuration settings for oauth relier
+    const integration = new OAuthIntegration(data, storageData, config.oauth);
+    this.initIntegration(integration);
+    this.initOAuthRelier(integration, this.flags);
+    this.initClientInfo(integration);
+    return integration;
+  }
+
+  private createDefaultIntegration(data: ModelDataStore) {
+    const integration = new WebIntegration(data);
+    this.initIntegration(integration);
+    return integration;
+  }
 
-    // TEMPORARY for tests until `sameBrowserVerificationContext` is accounted for.
-    // in reality this will never have this context, it'll get pulled from localStorage
-    if (this.flags.searchParam('context') === Constants.FX_DESKTOP_V3_CONTEXT) {
-      return new SyncDesktopIntegration();
+  private createSyncBasicIntegration(data: ModelDataStore) {
+    const integration = new SyncBasicIntegration(data);
+    this.initIntegration(integration);
+    return integration;
+  }
+
+  private createSyncDesktopIntegration(data: ModelDataStore) {
+    const integration = new SyncDesktopIntegration(data);
+    this.initIntegration(integration);
+    return integration;
+  }
+
+  /**
+   * Initializes a base relier state
+   **/
+  initIntegration(integration: Integration) {
+    // Important!
+    // FxDesktop declares both `entryPoint` (capital P) and
+    // `entrypoint` (lowcase p). Normalize to `entrypoint`.
+    const entryPoint = integration.data.getModelData().get('entryPoint');
+    const entrypoint = integration.data.getModelData().get('entrypoint');
+    if (
+      entryPoint != null &&
+      entrypoint != null &&
+      typeof entryPoint === 'string'
+    ) {
+      integration.data.entrypoint = entryPoint;
     }
-    return new WebIntegration();
   }
 
-  // TODO in FXA-7308
-  // private _getSameBrowserVerificationModel(namespace) {
```

Comment:
We're not accounting for this logic due to moving away from email verification links, [see more info](https://docs.google.com/document/d/1K4AD69QgfOCZwFLp7rUcMOkOTslbLCh7jjSdR9zpAkk/edit#heading=h.kkt4eylho93t).